doc/sphinx/source/libCEEDdev.md

```diff
@@ -88,10 +88,45 @@ and
 are purely implicit -- one just indexes the same array using the
 appropriate convention.
 
-## `restrict` semantics
+## `restrict` Semantics
 
 QFunction arguments can be assumed to have `restrict` semantics. That is, each input and output array must reside in distinct memory without overlap.
 
+## CeedVector Array Access Semantics
```

Comment:
I tried to distill the changes and discussion into the documentation here. These are the rules I have coded vector access to follow in this branch.

Reply:
I think this is very clear, thank you.

Reply:
Thanks for your help and review on this PR - I think we have some nice updates